feature/sdk-nativa (#2612)
mercadopago/devsite-docs: guides/snippets/api-mobile/ios.es.md

## Requisitos

Antes de comenzar la integración, asegúrate de que tu proyecto cumpla con los siguientes requisitos:

| Requisitos | Descripción |
|-|-|
| iOS | Versión 13 o superior |
| XCode | Versión 16 o superior |
| Swift | Versión 5.5 o superior |

> | Public Key | La :toolTipComponent[*_Public Key_*]{content="Clave pública que se utiliza en el _frontend_ para acceder a la información y cifrar datos. Puedes acceder a ella a través de *Tus integraciones > Detalles de la aplicación > Producción > Credenciales de producción*."} está directamente vinculada a la :toolTipComponent[aplicación]{link="/developers/es/docs/your-integrations/application-details" linkText="Detalles de la aplicación" content="Entidad registrada en Mercado Pago que actúa como un identificador para gestionar tus integraciones. Para más información, accede al enlace abajo."} que creaste, teniendo en cuenta que cada clave es única para cada integración. |

Reviewer: ```suggestion
| Public Key | La :toolTipComponent[*_Public Key_*]{content="Clave pública que se utiliza en el _frontend_ para acceder a la información y cifrar datos. Puedes acceder a ella a través de *Tus integraciones > Detalles de la aplicación > Producción > Credenciales de producción*."} está directamente vinculada a la :toolTipComponent[aplicación]{link="/developers/es/docs/your-integrations/application-details" linkText="Detalles de la aplicación" content="Entidad registrada en Mercado Pago que actúa como un identificador para gestionar tus integraciones. Para más información, accede al enlace abajo."} que creaste, por lo tanto, cada clave es única para cada integración. |
```

```swift
Task {
   // Se debe usar el bin retornado por OnBinChanged de CardNumberTextfield
    let paymentMethod = try await
coreMethods.paymentMethods(
bin: "5024111"
    )
}
```

Los parámetros están listados en la tabla a continuación.

| Parámetro | Tipo   | Descripción                                                                                                  | Obligatorio | 
|-----------|--------|-------------------------------------------------------------------------------------------------------------|-------------| 
| bin       | String | Los primeros 8 dígitos de la tarjeta de crédito, obtenidos por el _callback_ `onBinChange` de `CardNumberTextField`. | Obligatorio | 

Para más información sobre la respuesta de la llamada, consulta la [documentación del método en GitHub](https://mercadopago.github.io/sdk-ios/latest/documentation/coremethods/coremethods/paymentmethods(bin:mode:)).

:::
:::AccordionComponent{title="Obtener condiciones de ----[mla, mlb, mco, mlc, mlu, mpe]----cuotas----------------[mlm]----mensualidades------------" pill="client-side"} 

> El método Obtener condiciones de ----[mla, mlb, mco, mlc, mlu, mpe]----cuotas----------------[mlm]----mensualidades------------ te permite realizar la búsqueda de todas las opciones de ----[mla, mlb, mco, mlc, mlu, mpe]----cuotas----------------[mlm]----mensualidades------------ disponibles para una determinada tarjeta y monto de transacción. Considera las reglas del emisor, del método de pago y del valor de la compra, devolviendo todas las opciones válidas de ----[mla, mco, mlb, mlc, mlu, mpe]----cuotas----------------[mlm]----mensualidades------------, incluyendo cantidad de ----[mla, mlb, mco, mlc, mlu, mpe]----cuotas----------------[mlm]----mensualidades------------, intereses, valor de cada cuota, valor total, entre otros.

Reviewer: ```suggestion
Este método te permite realizar la búsqueda de todas las opciones de ----[mla, mlb, mco, mlc, mlu, mpe]----cuotas----------------[mlm]----mensualidades------------ disponibles para una determinada tarjeta y monto de transacción. Considera las reglas del emisor, del método de pago y del valor de la compra, devolviendo todas las opciones válidas de ----[mla, mco, mlb, mlc, mlu, mpe]----cuotas----------------[mlm]----mensualidades------------, incluyendo cantidad de ----[mla, mlb, mco, mlc, mlu, mpe]----cuotas----------------[mlm]----mensualidades------------, intereses, valor de cada cuota, valor total, entre otros.
```

Reviewer: Lo eliminaría para evitar la repetición del método con el título y de cuotas/mensualidades con lo que sigue en la oración.

| `expirationDateState` | [ExpirationDateTextfield](https://mercadopago.github.io/sdk-ios/latest/documentation/coremethods/expirationdatetextfield) | Clase del campo de expiración de la tarjeta.      | Obligatorio |
| `securityCodeState`   | [SecurityCodeTextField](https://mercadopago.github.io/sdk-ios/latest/documentation/coremethods/securitycodetextfield) | Clase del campo de código de seguridad de la tarjeta.   | Obligatorio |
| `documentType`        | [IdentificationType](https://mercadopago.github.io/sdk-ios/latest/documentation/coremethods/identificationtype) | Tipo de documento que se está enviando | Opcional    |
| `documentNumber`      | String             | Número del documento                     | Opcional    |
| `cardHolderName`      | String             | Nombre completo del titular de la tarjeta     | Obligatorio |

Para más información sobre la respuesta de la llamada, consulta la [documentación del método en GitHub](https://mercadopago.github.io/sdk-ios/latest/documentation/coremethods/coremethods/createtoken(cardnumber:expirationdate:securitycode:documenttype:documentnumber:cardholdername:)).

### Crear un _token_ para una tarjeta existente

> En las transacciones con Mercado Pago, los datos de las tarjetas registradas por el comprador se almacenan de forma segura y no son accesibles desde tu _backend_. Sólo se proporciona el `ID` de la tarjeta a la aplicación, que deberás utiliziar para generar un _token_ temporal. Esto protege la información sensible, ya que sólo se maneja el `ID`, mientras que el número de tarjeta, CVV y fecha de vencimiento no se exponen y se mantienen seguros.

Reviewer: ```suggestion
En las transacciones con Mercado Pago, los datos de las tarjetas registradas por el comprador se almacenan de forma segura y no son accesibles desde tu _backend_. Sólo se proporciona el `ID` de la tarjeta a la aplicación, que deberás utilizar para generar un _token_ temporal. Esto protege la información sensible, ya que sólo se maneja el `ID`, mientras que el número de tarjeta, CVV y fecha de vencimiento no se exponen y se mantienen seguros.
```

## Requisitos

> Antes de comenzar la integración, asegúrate de que tu proyecto cumple con los siguientes requisitos:

Reviewer: ```suggestion
Antes de comenzar la integración, asegúrate de que tu proyecto cumpla con los siguientes requisitos:
```

| Requisitos | Descripción |
|-|-|
| iOS | Versión 13 o superior |
| XCode | Versión 16 o superior |
| Swift | Versión 5.5 o superior |

> | Public Key | La :toolTipComponent[*_Public Key_*]{content="Clave pública que se utiliza en el _frontend_ para acceder a la información y cifrar datos. Puedes acceder a ella a través de *Tus integraciones > Detalles de la aplicación > Producción > Credenciales de producción*."} está directamente vinculada a la :toolTipComponent[aplicación]{link="/developers/es/docs/your-integrations/application-details" linkText="Detalles de la aplicación" content="Entidad registrada en Mercado Pago que actúa como un identificador para gestionar tus integraciones. Para más información, accede al enlace abajo."} que creaste, por lo tanto, cada una es única para cada integración. |

Reviewer: ```suggestion
| Public Key | La :toolTipComponent[*_Public Key_*]{content="Clave pública que se utiliza en el _frontend_ para acceder a la información y cifrar datos. Puedes acceder a ella a través de *Tus integraciones > Detalles de la aplicación > Producción > Credenciales de producción*."} está directamente vinculada a la :toolTipComponent[aplicación]{link="/developers/es/docs/your-integrations/application-details" linkText="Detalles de la aplicación" content="Entidad registrada en Mercado Pago que actúa como un identificador para gestionar tus integraciones. Para más información, accede al enlace abajo."} que creaste, teniendo en cuenta que cada clave es única para cada integración. |
```

## Configurar campos seguros

> Los campos seguros son componentes desarrollados para garantizar la privacidad y la protección de los datos sensibles introducidos por el comprador. En total conformidad con los estándares PCI, estos campos aseguran que la aplicación nunca tenga acceso directo a la información ingresada, la cual se transmite de forma segura solo para la creación de tokens y transacciones.

Reviewer: _Sugerencia_ NTH: podemos crear un tooltip con la definición de los estándares PCI

> Los campos seguros son componentes desarrollados para garantizar la privacidad y la protección de los datos sensibles introducidos por el comprador. En total conformidad con los estándares PCI, estos campos aseguran que la aplicación nunca tenga acceso directo a la información ingresada, la cual se transmite de forma segura solo para la creación de tokens y transacciones.

Reviewer: ```suggestion
Los campos seguros son componentes desarrollados para garantizar la privacidad y la protección de los datos sensibles introducidos por el comprador. En total conformidad con los estándares PCI, estos campos aseguran que la aplicación nunca tenga acceso directo a la información ingresada, la cual se transmite de forma segura solo para la creación de _tokens_ y transacciones.
```

Todas las interacciones con estos campos ocurren mediante _callbacks_, permitiendo la captura de eventos relevantes sin exponer los datos del usuario. Los métodos descritos a continuación utilizan instancias de estos campos seguros, por lo que es esencial que estén debidamente configurados en la interfaz del checkout antes de utilizarlos.

Consulta en la tabla a continuación los componentes disponibles. Para más detalles sobre la configuración, consulta la referencia correspondiente en GitHub.

| Nombre del Componente        | Referencia en GitHub                                                                                                                 | Descripción                                              |
|-----------------------------|--------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------|

> | CardNumberTextField         | [Referencia](https://mercadopago.github.io/sdk-ios/0.1.0/documentation/coremethods/cardnumbertextfield)                              | Campo seguro para ingresar el número de la tarjeta.      |

Reviewer: Gabay, le di formato de código, pero si crees que no corresponde porque son componentes y no parámetros, sacalos.

| ExpirationDateTextField     | [Referencia](https://mercadopago.github.io/sdk-ios/0.1.0/documentation/coremethods/expirationdatetextfield)                          | Campo seguro para ingresar la fecha de vencimiento.      |

> | SecurityTextField           | [Referencia](https://mercadopago.github.io/sdk-ios/0.1.0/documentation/coremethods/securitycodetextfield)                            | Campo seguro para ingresar el código de seguridad (CVV). |

Reviewer: ```suggestion
| `SecurityTextField`           | [Referencia](https://mercadopago.github.io/sdk-ios/0.1.0/documentation/coremethods/securitycodetextfield)                            | Campo seguro para ingresar el código de seguridad (CVV). |
```

Cada componente notifica a la aplicación cada vez que haya un cambio de valor, sin exponer los datos introducidos, e informa también el resultado de la validación del campo según las reglas de PCI y de la tarjeta.

> NOTE
>
> Los datos introducidos en los campos seguros nunca están disponibles para la aplicación integradora. Se envían de forma segura solo para la creación de _tokens_ y transacciones.

> ## _Core Methods_

Reviewer: ```suggestion
## Core Methods
```

Los **_Core Methods_** son esenciales para construir un flujo de checkout integrado con Mercado Pago. Utilizan la información capturada por los [**campos seguros**](#bookmark_campos_seguros) y permiten ejecutar las principales operaciones de pago.

Cada método debe ser utilizado según las necesidades de tu flujo de pago. Para utilizarlos, comienza creando una instancia de _Core Methods_ en tu clase utilizando el siguiente código:

```swift
private let coreMethods = CoreMethods() 
```

Así podrás utilizar cualquiera de los métodos listados a continuación:

:::AccordionComponent{title="Obtener medios de pago" pill="client-side"} 

> El método **Obtener medios de pago** devuelve la lista de medios de pago disponibles en función del BIN de la tarjeta informada, considerando las reglas e instituciones financieras válidas para el país configurado. Este paso es esencial para identificar la marca de la tarjeta, configurar correctamente los siguientes pasos del checkout y validar la aceptación de la tarjeta.

Reviewer: Gabay, muchos de los ajustes o sugerencias que marqué arriba se replicarían en las definiciones de acá :) asi que no las voy a marcar de nuevo para no repetir tanto.
